zeeguu/core/word_scheduling/basicSR/basicSR.py

```diff
@@ -46,19 +48,40 @@ def __init__(self, bookmark=None, bookmark_id=None):
         self.next_practice_time = datetime.now()
         self.consecutive_correct_answers = 0
         self.cooling_interval = 0
+    
+    def set_bookmark_as_learned(self, db_session):
+        self.bookmark.learned = True
+        self.bookmark.learned_time = datetime.now()
+        db_session.add(self.bookmark)
+        db_session.commit()
+        db_session.delete(self)
+        db_session.commit()
 
     def update_schedule(self, db_session, correctness):
+        learning_cycle = self.bookmark.learning_cycle
+
+        # Query user preferences to get the value of productive exercises setting
+        productive_exercises_setting = UserPreference.query.filter_by(user_id=self.bookmark.user_id, key="productive_exercises").first().value
+
         if correctness:
-            # Use the cooldown to check if the user has learned
-            # the word.
             if self.cooling_interval == MAX_INTERVAL_8_DAY:
-                self.bookmark.learned = True
-                self.bookmark.learned_time = datetime.now()
-                db.session.add(self.bookmark)
-                db.session.commit()
-                db.session.delete(self)
-                db.session.commit()
-                return
+                if learning_cycle == LearningCycle.NOT_SET:
+                    self.set_bookmark_as_learned(db_session)
```

Comment:
I think here you can reverse the conditions maybe for simpler code? I.e. if `learningCycle == RECEPTIVE and setting=="true"` then do the switch of the learning cycle, otherwise for all the other cases we seem to be doing the "`set_bookmark_as_learned(db_session)`". Or?